### PR TITLE
Entire database connection is established as per the spring framework…

### PR DESCRIPTION
**Pull request by mjadhav1**
…. Error message will be shown if the database is not configured.
I have changed the workflow of database connectivity. This is a major change. Please review it and suggest if any changes needed.

### CODE REVIEW

**Comment by mjadhav1** [comment]
You will get the database credentials in the **spring-datasource.xml** file.

**Review by tobielf — CHANGES_REQUESTED** [review]
1. Do we need to change from `Customer` to `customer`? I saw a lot of files only change on this. Try to keep the changes as minimal as possible. 
2. In your java code, unify your indentation; currently, your code is mixing space and tab.
3. Add the comment to all newly added code.

**Comment by mjadhav1** [comment]
@tobielf for 2. Can you tell me which java file has mixing of tabs and space? 
for 3. I have already added comments to the new code. Please check UserController.java, UserService.java, UserDAO.java

**Comment by mjadhav1** [comment]
@tobielf For 1. I have reverted the bower_components folder with the master. Now everything is back to the "Customer" from "customer". Please review it and let me know.

File: src/main/java/edu/asu/plp/controller/UserController.java [hunk 1]
```diff
@@ -19,12 +13,27 @@
 	
 	@RequestMapping("/user")
 	public Principal userLogin(Principal principal){
-		return principal;
+        return principal;
```
Comment (tobielf) [inline src/main/java/edu/asu/plp/controller/UserController.java]: Tab indent here.

File: src/main/java/edu/asu/plp/controller/UserController.java [hunk 2]
```diff
@@ -19,12 +13,27 @@
...
-	public void saveUser(@RequestBody UserModel user){
-		
-		userService.saveUser(user.email, user.firstName, user.lastName);
+
+    /**
+     * @brief This method checks if the database is configured properly or not
+     * @return true if the database is configured properly otherwise false
+     */
+	@RequestMapping("/checkDBConnection")
+    public boolean checkDBConnection() {
```
Comment (tobielf) [inline src/main/java/edu/asu/plp/controller/UserController.java]: Tab indent here.

File: src/main/java/edu/asu/plp/service/UserService.java [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
+
+    UserDAO userDAO = (UserDAO) context.getBean("userDAO");
+
+    /***
+     * @brief This method save the user information in the database
+     * @param user
+     * @return string as success or error
+     */
+    public String saveUser(User user){
+        return userDAO.saveUser(user);
```
Comment (tobielf) [inline src/main/java/edu/asu/plp/service/UserService.java]: Tab indent here.

File: src/main/java/edu/asu/plp/user/dao/impl/JdbcUserDAO.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import edu.asu.plp.user.dao.UserDAO;
+import edu.asu.plp.user.model.User;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class JdbcUserDAO implements UserDAO {
```
Comment (tobielf) [inline src/main/java/edu/asu/plp/user/dao/impl/JdbcUserDAO.java]: Comment your code.

File: target/classes/static/JS/plpController.js [hunk 5]
```diff
@@ -8,22 +8,41 @@ app.controller('idectrl', [ '$scope', '$cookies', '$http', function( $scope, $co
...
+        });
 	};
-	getUser();
+
+    /***
+     * @brief This method checks if the database is configured properly or not
+     * If the database is not configured properly then the alert will be displayed otherwise user will be redirected
+     * to the main layout page
+     */
+	var checkDatabaseConnection = function () {
```
Comment (tobielf) [inline target/classes/static/JS/plpController.js:38]: Check your indent.

**Review by tobielf — COMMENTED** [review]
This protect method, should not be checkDatabaseConnection() and invoke at the start-up. Instead, you should handle the AJAX return value, like in  `plpController.js`, line 23, the response contains the error message, and alert to the user the detailed message so that it can be generalized. If there is no MySQL, warning on: configure a MySQL. If there is no DB named plp_db, warning on creating a DB. If there is no `user_info` table, warning on this type error. Error message needs provide enough, detail information, and the checking should only happen when an error occurred. 

In your current implementation, this function will invoke every time when the page is ready, which will impact the performance when the system is fine.

**Review by tobielf — APPROVED** [review]

**Pull request merged**